fasthttp_http_client_factory.go

```diff
@@ -24,6 +25,23 @@ func (*fasthttpHttpClientFactory) Create(o httpClientOptions) httpClient {
 		d = fasthttpproxy.FasthttpHTTPDialerTimeout(o.Proxy, tcpTimeout)
 	}
 
+	if o.CustomDnsAddr != "" {
+		d = func(addr string) (net.Conn, error) {
+			dialer := fasthttp.TCPDialer{
+				Concurrency: 1000,
+				Resolver: &net.Resolver{
+					PreferGo:     true,
```

Comment:
Is there any particular reason you enabled this option?

Reply:
i think if we don't enable this option, it uses the DNS resolver of our operating system, I tried sniffing the DNS packets through wireshark, the custom dns resolver was used only if I turn on this flag.

Reply:
have a look at https://github.com/golang/go/issues/35561#issuecomment-553621030

Reply:
thanks! I reverted it back on the `master` branch.

Reply:
Sure will do and get back to you !

Reply:
hey @raviqqe just tested it, seems to be working fine

![image](https://github.com/user-attachments/assets/bde8fb60-995a-4f67-a8f9-65765918bb65)

Reply:
Thanks!